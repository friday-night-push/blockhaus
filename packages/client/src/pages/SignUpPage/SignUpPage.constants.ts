import * as Yup from 'yup';

import { InputProps } from 'src/components/atoms';

const phoneRegExp =
  /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;

export const inputs: InputProps[] = [
  { name: 'email', label: 'Email', type: 'email', autoComplete: 'email' },
  {
    name: 'first_name',
    label: 'First Name',
    type: 'text',
    autoComplete: 'given-name',
  },
  {
    name: 'second_name',
    label: 'Second Name',
    type: 'text',
    autoComplete: 'family-name',
  },
  { name: 'login', label: 'Login', type: 'text', autoComplete: 'username' },
  {
    name: 'password',
    label: 'Password',
    type: 'password',
    autoComplete: 'new-password',
  },
  { name: 'phone', label: 'Phone', type: 'tel', autoComplete: 'tel' },
];

export const validationSchema = Yup.object({
  email: Yup.string()
    .email('Invalid email address')
    .required('Email is required'),
  first_name: Yup.string()
    .max(50, 'First name must be 50 characters or less')
    .required('First name is required'),
  second_name: Yup.string()
    .max(50, 'Second name must be 50 characters or less')
    .required('Second name is required'),
  login: Yup.string()
    .max(15, 'Login must be 15 characters or less')
    .required('Login is required'),
  password: Yup.string()
    .min(8, 'Password must be at least 8 characters')
    .required('Password is required'),
  phone: Yup.string()
    .matches(phoneRegExp, 'Phone number is not valid')
    .required('Phone number is required'),
});
