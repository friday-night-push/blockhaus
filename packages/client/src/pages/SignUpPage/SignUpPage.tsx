import { useEffect, useState } from 'react';
import { TSignUpRequest, TUser } from '../../shared/types/user';
import AuthAPI from '../../services/api/auth-api';
import { Link } from 'react-router-dom';

const authAPI = new AuthAPI();

export const SignUpPage = () => {
  const [user, setUser] = useState<TUser>({} as TUser);
  const [data, setData] = useState<TSignUpRequest>({} as TSignUpRequest);

  useEffect(() => {
    const login = Math.floor(Math.random() * 1000);

    const _data: TSignUpRequest = {
      email: `a${login}@mail.mail`,
      first_name: `Fname_${login}`,
      second_name: `Sname_${login}`,
      login: `l${login}`,
      password: '123!!321',
      phone: `+79994447${login}`,
    };

    setData(_data);

    authAPI.signup(_data, updUserData, errorHandler);
  }, []);

  const updUserData = (u: TUser) => {
    setUser(u);
  };

  const errorHandler = (err: unknown) => {
    console.error('EEEEEEEEEEEE', err);
  };

  function logout(): void {
    authAPI.logout(logoutHandler, errorHandler);
  }

  const logoutHandler = () => {
    setUser({} as TUser);
  };

  return (
    <>
      <div>SignUpPage</div>
      <hr />
      <Link to="/">Return to login</Link>
      <hr />
      <div>
        <strong>New user data</strong>
      </div>
      <pre>{JSON.stringify(data, null, '    ')}</pre>
      <hr />
      <div>
        <strong>Register user data</strong>
      </div>
      <pre>{JSON.stringify(user, null, '    ')}</pre>
      <hr />
      <button onClick={logout}>Logout</button>
    </>
  );
};
