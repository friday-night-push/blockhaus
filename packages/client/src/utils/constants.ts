export const HOST = 'http://friday-night-push-blockhau-40.ya-praktikum.tech';
export const SERVER_HOST_URL = `${HOST}:3001`;
export const API_URL = `${SERVER_HOST_URL}/api/v1`;
export const YANDEX_URL = `${API_URL}/proxy`;

// OAuth
export const REDIRECT_URL = `${HOST}:3000`;

export const RESOURCE_URL = YANDEX_URL + '/resources';

export enum PAGE_ROUTES {
  MENU = '/',
  SIGN_IN = '/sign-in',
  SIGN_UP = '/sign-up',
  CHANGE_PASSWORD = '/change-password',
  RECOVER_PASSWORD = '/recover-password',
  PROFILE = '/profile',
  GAME = '/game',
  GAME_PAUSE = '/game-pause',
  GAME_RESULT = '/game-result',
  GAME_SETUP = '/game-setup',
  GAME_DIFFICULT = '/game-difficult',
  LEADER_BOARD = '/leader-board',
  ERROR_BOUNDARY = '/error-boundary',
  FORUM = '/forum',
  FORUM_TOPIC = '/forum/:topicId/:page?',
}
