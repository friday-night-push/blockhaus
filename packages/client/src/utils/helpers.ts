// Отдельный класс исключений для ESLint

export default class Helpers {
  public static Log(level: string, ...args: any): any {
    switch (level) {
      case 'INFO':
        // eslint-disable-next-line no-console
        console.info(...args);
        break;
      case 'LOG':
        // eslint-disable-next-line no-console
        console.log(...args);
        break;
      case 'ERROR':
        // eslint-disable-next-line no-console
        console.error(...args);
        break;
      default:
        break;
    }
  }
}
