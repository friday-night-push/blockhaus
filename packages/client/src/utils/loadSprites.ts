export function loadSprites(
  spritesPath: Record<string, string> = {}
): Promise<Record<string, HTMLImageElement>> {
  return Promise.all(
    Object.values(spritesPath).map(path => loadImage(path))
  ).then(sprites => {
    const keys = Object.keys(spritesPath);
    const map = keys.map((key, index) => [key, sprites[index]]);
    return Object.fromEntries(map);
  }) as Promise<Record<string, HTMLImageElement>>;
}

function loadImage(src: string): Promise<HTMLImageElement> {
  return new Promise<HTMLImageElement>((resolve, reject) => {
    const img = new Image();
    img.onload = () => resolve(img);
    img.onerror = reject;
    img.src = src;
  });
}
