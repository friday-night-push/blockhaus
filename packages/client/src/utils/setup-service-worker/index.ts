export * from './setup-service-worker';
