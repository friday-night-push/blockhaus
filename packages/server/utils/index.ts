export * from './init-postgres';
export * from './is-dev';
export * from './logger';
