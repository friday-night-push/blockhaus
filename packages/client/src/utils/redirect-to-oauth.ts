import { REDIRECT_URL } from './constants';

export const redirectToOAuth = (serviceId: string) => {
  window.location.replace(
    `https://oauth.yandex.ru/authorize?response_type=code&client_id=${serviceId}&redirect_uri=${REDIRECT_URL}`
  );
};
