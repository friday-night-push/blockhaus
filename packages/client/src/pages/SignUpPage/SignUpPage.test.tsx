import userEvent from '@testing-library/user-event';

import { PAGE_ROUTES } from 'src/utils/constants';
import { render, screen } from 'src/utils/tests';

import { mockedUseNavigate, mockedUser } from 'src/utils/tests/mocks';

import { SignUpPage } from './SignUpPage';

describe('SignUpPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot', () => {
    const { container } = render(<SignUpPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('navigate to menu if user is authenticated', async () => {
    render(<SignUpPage />, {
      authState: {
        user: mockedUser,
        userIsLoading: false,
      },
    });
    expect(screen.queryByText('Sign up')).not.toBeInTheDocument();
  });

  it('navigates back on back button click', async () => {
    render(<SignUpPage />);

    const user = userEvent.setup();
    await user.click(screen.getByRole('button', { name: 'Back' }));
    expect(mockedUseNavigate).toHaveBeenCalledWith(-1);
  });

  it('navigates to sign in on sign in button click', async () => {
    render(<SignUpPage />);

    const user = userEvent.setup();
    await user.click(
      screen.getByRole('button', { name: 'Signed up already? Sign In' })
    );
    expect(mockedUseNavigate).toHaveBeenCalledWith(PAGE_ROUTES.SIGN_IN);
  });
});
