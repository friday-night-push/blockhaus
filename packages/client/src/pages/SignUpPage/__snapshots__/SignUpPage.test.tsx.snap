// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpPage matches snapshot 1`] = `
<section
  class="g-box g-flex"
  style="width: auto; min-height: 100vh; column-gap: 12px; row-gap: 12px; justify-content: center;"
>
  <section
    class="g-box g-flex"
    style="width: 100%; min-height: 100vh; max-width: 380px; flex-direction: column; column-gap: 32px; row-gap: 32px; align-items: center; justify-content: center;"
  >
    <img
      alt="Blockhaus"
      style="display: flex; width: 240px;"
    />
    <section
      class="g-box g-flex"
      style="flex-direction: column; column-gap: 12px; row-gap: 12px; align-items: center;"
    >
      <span
        class="g-text g-text_variant_display-2"
      >
        Sign up
      </span>
      <span
        class="g-text g-text_variant_subheader-2"
        style="text-align: center;"
      >
        New here? Dive in! Just fill in the form and let the good times roll.
      </span>
      <form
        action="#"
        style="width: 100%;"
      >
        <section
          class="g-box g-flex"
          style="width: 100%; flex-direction: column; column-gap: 16px; row-gap: 16px; align-items: center;"
        >
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:r0:"
                style="max-width: calc(50% - 0px);"
                title="Email"
              >
                Email
              </label>
              <input
                autocomplete="email"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:r0:"
                name="email"
                type="email"
                value=""
              />
            </span>
          </span>
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:r3:"
                style="max-width: calc(50% - 0px);"
                title="First Name"
              >
                First Name
              </label>
              <input
                autocomplete="given-name"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:r3:"
                name="first_name"
                type="text"
                value=""
              />
            </span>
          </span>
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:r6:"
                style="max-width: calc(50% - 0px);"
                title="Second Name"
              >
                Second Name
              </label>
              <input
                autocomplete="family-name"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:r6:"
                name="second_name"
                type="text"
                value=""
              />
            </span>
          </span>
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:r9:"
                style="max-width: calc(50% - 0px);"
                title="Login"
              >
                Login
              </label>
              <input
                autocomplete="username"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:r9:"
                name="login"
                type="text"
                value=""
              />
            </span>
          </span>
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:rc:"
                style="max-width: calc(50% - 0px);"
                title="Password"
              >
                Password
              </label>
              <input
                autocomplete="new-password"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:rc:"
                name="password"
                type="password"
                value=""
              />
            </span>
          </span>
          <span
            class="g-text-input g-text-input_view_normal g-text-input_size_xl g-text-input_pin_brick-brick"
          >
            <span
              class="g-text-input__content"
            >
              <label
                class="g-text-input__label"
                for="g-:rf:"
                style="max-width: calc(50% - 0px);"
                title="Phone"
              >
                Phone
              </label>
              <input
                autocomplete="tel"
                class="g-text-input__control g-text-input__control_type_input"
                id="g-:rf:"
                name="phone"
                type="tel"
                value=""
              />
            </span>
          </span>
          <section
            class="g-box g-flex"
            style="flex-direction: row; column-gap: 8px; row-gap: 8px;"
          >
            <button
              class="g-button g-button_view_action g-button_size_xl g-button_pin_brick-brick g-button_width_max"
              type="submit"
            >
              <span
                class="g-button__text"
              >
                Submit
              </span>
            </button>
          </section>
        </section>
      </form>
      <a
        aria-disabled="false"
        class="g-button g-button_view_flat g-button_size_xl g-button_pin_brick-brick"
        href="/sign-in"
      >
        <span
          class="g-button__text"
        >
          Signed up already? Sign In
        </span>
      </a>
      <button
        class="g-button g-button_view_flat g-button_size_xl g-button_pin_brick-brick"
        type="button"
      >
        <span
          class="g-button__text"
        >
          Go Back
        </span>
      </button>
    </section>
  </section>
</section>
`;
